vector/src/main/java/im/vector/app/features/location/LocationSharingViewState.kt

```diff
@@ -39,7 +41,8 @@ data class LocationSharingViewState(
 
     constructor(locationSharingArgs: LocationSharingArgs) : this(
             roomId = locationSharingArgs.roomId,
-            mode = locationSharingArgs.mode
+            mode = locationSharingArgs.mode,
+            isLiveLocationSharingEnabled = BuildConfig.ENABLE_LIVE_LOCATION_SHARING
```

Comment:
Maybe a stupid question but should that be reading from `DebugVectorFeatures` instead?

Reply:
This is a good question but here, unfortunately we cannot use dependency injection to create the corresponding `VectorFeatures` implementation. Another solution would be to declare the boolean as nullable and initialize it to `null`. When `null`, we would consider the feature as disabled. Do you think it is a better approach?

Reply:
Oh, if that's not possible technically, that's fine. I only meant to verify whether or not usage of the build config flag was intentional here.

Reply:
we could mark this as `nullable` rather than mix the build config and feature flags, what do you think?

thinking out loud, it would be great if the assisted viewmodel initial state could use the dagger graph something like 

```kotlin
@Provides fun provideInitial(features: VectorFeatures) = LocationSharingViewState(
  features.isLiveLocationEnabled()
)
```

not sure if this is possible with our current setup~

Reply:
I removed the parameter from the state to simplify.